core/time/timezone/tzdate.odin

```diff
@@ -0,0 +1,320 @@
+
+NOTE: tz will be referenced in the result datetime, so it must stay alive/allocated as long as it is used
+Returns:
+- out: The converted datetime
+- success: `false` if the datetime was invalid
+*/
+datetime_to_tz :: proc(dt: datetime.DateTime, tz: ^datetime.TZ_Region) -> (out: datetime.DateTime, success: bool) #optional_ok {
+	dt := dt
+	if dt.tz == tz {
+		return dt, true
+	}
+	if dt.tz != nil {
+		dt = datetime_to_utc(dt)
+	}
+	if tz == nil {
+		return dt, true
+	}
+
+	tm := time.datetime_to_time(dt) or_return
+	record := region_get_nearest(tz, tm) or_return
+
+	secs := time.time_to_unix(tm)
+	adj_time := time.unix(secs + record.utc_offset, 0)
+	adj_dt := time.time_to_datetime(adj_time) or_return
+	adj_dt.tz = tz
+
+	return adj_dt, true
+}
+
+/*
+Gets the timezone abbreviation/shortname for a given date.
+(ex: "PDT")
+
+Inputs:
+- dt: The datetime containing the date, time, and timezone pointer for the lookup
+
+NOTE: The lifetime of name matches the timezone it was pulled from.
+Returns:
+- name: The timezone abbreviation
+- success: returns `false` if the passed datetime is invalid
+*/
+shortname :: proc(dt: datetime.DateTime) -> (name: string, success: bool) #optional_ok {
+	tm := time.datetime_to_time(dt) or_return
+	if dt.tz == nil { return "UTC", true }
+
+	record := region_get_nearest(dt.tz, tm) or_return
+	return record.shortname, true
+}
+
+/*
+Gets the timezone abbreviation/shortname for a given date.
+(ex: "PDT")
+
+WARNING: This is unsafe because it doesn't check if you datetime is valid, or your region contains a valid record
+
+Inputs:
+- dt: The input datetime
+
+NOTE: The lifetime of name matches the timezone it was pulled from.
+Returns:
+- name: The timezone abbreviation
+*/
+shortname_unsafe :: proc(dt: datetime.DateTime) -> string {
+	if dt.tz == nil { return "UTC" }
+
+	tm, _ := time.datetime_to_time(dt)
+	record, _ := region_get_nearest(dt.tz, tm)
+	return record.shortname
+}
+
+/*
+Checks DST for a given date.
+
+Inputs:
+- dt: The input datetime
+
+Returns:
+- is_dst: returns `true` if dt is in daylight savings time, `false` if not
+- success: returns `false` if the passed datetime is invalid
+*/
+dst :: proc(dt: datetime.DateTime) -> (is_dst: bool, success: bool) #optional_ok {
+	tm := time.datetime_to_time(dt) or_return
+	if dt.tz == nil { return false, true }
+
+	record := region_get_nearest(dt.tz, tm) or_return
+	return record.dst, true
+}
+
+/*
+Checks DST for a given date.
+
+WARNING: This is unsafe because it doesn't check if you datetime is valid, or your region contains a valid record
+
+Inputs:
+- dt: The input datetime
+
+Returns:
+- is_dst: returns `true` if dt is in daylight savings time, `false` if not
+*/
+dst_unsafe :: proc(dt: datetime.DateTime) -> bool {
+	if dt.tz == nil { return false }
+
+	tm, _ := time.datetime_to_time(dt)
+	record, _ := region_get_nearest(dt.tz, tm)
+	return record.dst
+}
+
+datetime_to_str :: proc(dt: datetime.DateTime, allocator := context.temp_allocator) -> string {
```

Comment:
Should this actually default to the temp allocator? That may be confusing when so many APIs default to the normal allocator.

Reply:
I could switch these to aprintf, I suppose?
binding the temp_allocator so I can adjust what tprintf is using here

Reply:
The comment was more towards the `allocator := context.temp_allocator` instead of `allocator := context.allocator`, I think I would be confused when I call this function and it uses the temp allocator when most procedures default to the non-temp allocator instead.